service/src/main/java/bio/terra/workspace/service/resource/controlled/flight/clone/azure/database/CreateDbDumpEncryptionKeyStep.java

```diff
@@ -0,0 +1,47 @@
+package bio.terra.workspace.service.resource.controlled.flight.clone.azure.database;
+
+import bio.terra.stairway.FlightContext;
+import bio.terra.stairway.Step;
+import bio.terra.stairway.StepResult;
+import bio.terra.stairway.StepStatus;
+import bio.terra.stairway.exception.RetryException;
+import bio.terra.workspace.service.workspace.flight.WorkspaceFlightMapKeys;
+import java.security.NoSuchAlgorithmException;
+import java.security.SecureRandom;
+import javax.crypto.KeyGenerator;
+import javax.crypto.SecretKey;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class CreateDbDumpEncryptionKeyStep implements Step {
```

Comment:
this exists because it is theoretically possible for a malicious user to gain access to the database dump file. Because that file is an arbitrary psql script, it could be used to do all sorts of bad things.